lib/gen_socket_client.ex

```diff
@@ -321,16 +319,24 @@ defmodule Phoenix.Channels.GenSocketClient do
   end
 
   defp reinit(state) do
-    :ets.delete_all_objects(state.message_refs)
+    :erlang.erase()
```

Comment:
Good catch, we definitely shouldn't delete all other keys (which might include an RNG state for example). Your proposal sounds sensible. You can use`Process.get_keys/0` for that.